src/preview/hooks/useSource.js

```diff
@@ -0,0 +1,104 @@
+import React, { useState, useEffect, useLayoutEffect } from 'react';
+const useSource = ({ source, minWidth = 200, minHeight = 200 }) => {
+  const [isSource, setIsSource] = useState(false);
+  const [orientation, setOrientation] = useState('vertical');
```

Comment:
Might be for another ticket but, isSource and orientation reset whenever the story changes, would be nice to persist that somehow